File: court-booking-frontend/packages/types/src/waitlist.ts
export interface WaitlistJoinInput {
  court_id: string;
  slot_starts_at: string;
}

export interface WaitlistJoinOut {
  position: number;
}

export interface WaitlistEntry {
  id: string;
  court_id: string;
  court_name: string;
  venue_id: string;
  venue_name: string;
  player_id: string;
  slot_starts_at: string;
  position: number;
  notified_at: string | null;
  is_active: boolean;
  created_at: string;
}

File: court-booking-backend/app/schemas/waitlist.py
import uuid
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class WaitlistCreateIn(BaseModel):
    court_id: uuid.UUID
    slot_starts_at: datetime


class WaitlistJoinOut(BaseModel):
    position: int


class WaitlistOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    court_id: uuid.UUID
    court_name: str
    venue_id: uuid.UUID
    venue_name: str
    player_id: uuid.UUID
    slot_starts_at: datetime
    position: int
    notified_at: datetime | None
    is_active: bool
    created_at: datetime
